src/lexer/tokens/tokentype.rs

```diff
@@ -0,0 +1,87 @@
+#[derive(Debug, PartialEq, Eq)]
+pub enum TokenType {
+    // single character
+    LeftParen,
+    RightParen,
+    LeftBracket,
+    RightBracket,
+    LeftBrace,
+    RightBrace,
+    Plus,
+    Minus,
+    Slash,
+    BackSlash,
+    Star,
+    Modulus,
+    Pipe,
+    Ampersand,
+    Tilde,
+    Greater,
+    Lesser,
+    Comma,
+    Bang,
+    Dollar,
+    Assign,
+    Dot,
+    Colon,
+    Semicolon,
+    QuestionMark,
+    At,
+    Hash,
+    Quote,
+    DoubleQuote,
+    BackTick,
+    Caret,
+    Underscore,
+    NewLine,
+    Space,
+
+    // Double character
+    Equals,
+    BangEquals,
+    GreaterEquals,
+    LesserEquals,
+    And,
+    Or,
+    PlusEquals,
+    MinusEquals,
+    StarEquals,
+    SlashEquals,
+    ModEquals,
+    AmpersandEquals,
+    PipeEquals,
+    CaretEquals,
+    Increment,
+    Decrement,
+    RightShift,
+    LeftShift,
+    LeftArrow,
+    RightArrow,
+
+    // Literals
+    Identifier,
+    String,
+    Char,
+    Int,
+    Float,
+
+    // Registers
+    Register,
```

Comment:
why have two register types, maybe have Register(Type,Size)?

Reply:
I tried making a single enum variant was having little trouble with it. 
idk, why don't we stick to something simple and worry about optimizing this stuff later?

Reply:
not really. its just a couple of extra steps. the complexity wont increase

Reply:
Perhaps you didn't make the members public?
```rust
pub struct Register(pub u8, pub RegisterType);

pub enum RegisterType {
     ....
}
```